### ISSUE TITLE
[bug]: Calendar widget adds a date prior to the selected date

### ISSUE THREAD

**Issue by Huuuze** [makeplane/plane#3305]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When editing an issue, if you select a value for either Start Date or End Date, the value is the day prior to the selected date (i.e., I select "January 5, 2024", but January 4, 2024" is displayed instead).

https://github.com/makeplane/plane/assets/192399/64d66af5-4b92-4e48-8f51-037d58472396

### Steps to reproduce

See the screen-recording above.

This issue exists in Firefox 121 and Chromium 120.0.6099.144

### Browser

Mozilla Firefox

### Version

Self-hosted

**Comment by Huuuze** [comment]
I believe I've found the error. 

https://github.com/makeplane/plane/blob/preview/web/helpers/date-time.helper.ts#L186C22-L186C22

The method, "renderShortDate()", accepts a date string with this format: "2024-01-17". This value is cast into a Date object, but it does not include a time. The lack of a time value creates a problem here; the Date object appears to use the user's local timezone.

So, when using "Date.getDate()" to return the day value of the date string, the timezone ends up displaying the wrong day value. This code snippet demonstrates the problem; I believe the resolution is to use Date.getUTCDate().
```

const birthday = new Date('2024-01-17');
const non_utc_date = birthday.getDate();
const utc_date = birthday.getUTCDate();

// Expected output: 17
console.log(non_utc_date); //16
console.log(utc_date); //17

```

**Comment by Marinofull** [comment]
I got this same error, it is caused here because I'm in a negative timezone (gmt -3) so when it renders displays the wrong date. If I switch pc hour to a positive timezone, the bug disappear. It might be required a normalization of the date object before renders it on the screen.

**Comment by Marinofull** [comment]
I can confirm this bug is still present on v0.15.0

**Comment by jordan-zimmitti-newmile** [comment]
@Marinofull I can as well, hoping this well be fixed in the next release!

**Comment by ovvesley** [comment]
I can confirm this bug is still present on v0.16.0.

**Comment by pushya22** [comment]
This issue is resolved in PR #3992
